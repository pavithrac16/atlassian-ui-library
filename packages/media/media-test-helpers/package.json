{
  "name": "@atlaskit/media-test-helpers",
  "version": "25.0.4",
  "description": "Collection of test helpers used in media component stories and specs",
  "license": "Apache-2.0",
  "atlaskit:src": "src/index.tsx",
  "module": "dist/esm/index.js",
  "main": "dist/cjs/index.js",
  "types": "dist/cjs/index.d.ts",
  "author": "Atlassian Pty Ltd",
  "maintainers": [
    "Jimmy L",
    "Hector Z",
    "James N",
    "Aleksandr M",
    "Ali C",
    "Michael J"
  ],
  "atlaskit": {
    "internal": true,
    "name": "Media Test Helpers",
    "team": "media"
  },
  "repository": "https://bitbucket.org/atlassian/atlaskit-mk-2",
  "dependencies": {
    "@atlaskit/dropdown-menu": "^8.0.17",
    "@atlaskit/icon": "^19.0.6",
    "@atlaskit/media-card": "^64.0.2",
    "@atlaskit/media-client": "^2.0.5",
    "@atlaskit/media-core": "^30.0.13",
    "@atlaskit/media-picker": "^47.0.5",
    "@atlaskit/media-store": "^12.0.11",
    "@atlaskit/media-ui": "^11.5.5",
    "@atlaskit/select": "^10.0.8",
    "enzyme": "^3.7.0",
    "exenv": "^1.2.2",
    "kakapo": "^3.3.5",
    "lodash.matches": "^4.6.0",
    "react-intl": "^2.6.0",
    "react-redux": "^5.0.0",
    "styled-components": "^3.2.6",
    "tslib": "^1.9.3",
    "uuid": "^3.1.0",
    "xhr-mock": "^2.4.0"
  },
  "peerDependencies": {
    "react": "^16.8.0",
    "rxjs": "^5.5.0",
    "styled-components": "^3.2.6"
  },
  "devDependencies": {
    "rxjs": "^5.5.0"
  }
}