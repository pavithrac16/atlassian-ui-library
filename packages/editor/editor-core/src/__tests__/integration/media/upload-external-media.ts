import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import { editable, getDocFromElement, fullpage } from '../_helpers';
import {
  goToEditorTestingExample,
  mountEditor,
} from '../../__helpers/testing-example-helpers';
import { sleep } from '@atlaskit/media-test-helpers';

const baseADF = {
  version: 1,
  type: 'doc',
  content: [
    {
      type: 'mediaSingle',
      attrs: {
        layout: 'center',
      },
      content: [
        {
          type: 'media',
          attrs: {
            type: 'external',
            url:
              'https://images2.minutemediacdn.com/image/upload/c_crop,h_1193,w_2121,x_0,y_175/f_auto,q_auto,w_1100/v1554921998/shape/mentalfloss/549585-istock-909106260.jpg',
          },
        },
      ],
    },
    {
      type: 'paragraph',
      content: [],
    },
  ],
};

BrowserTestCase(
  'upload-external-media.ts: Uploads external media to media services',
  { skip: ['edge', 'ie', 'safari'] },
  async (client: any, testCase: string) => {
    const page = await goToEditorTestingExample(client);
    await mountEditor(page, {
      appearance: fullpage.appearance,
      defaultValue: JSON.stringify(baseADF),
      media: {
        allowMediaSingle: true,
      },
    });

    await page.waitForSelector('.ProseMirror .wrapper');
    await page.click('.ProseMirror .wrapper');
    await page.copy(editable);

    await page.paste(editable);

    await sleep(0);
    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testCase);
  },
);
