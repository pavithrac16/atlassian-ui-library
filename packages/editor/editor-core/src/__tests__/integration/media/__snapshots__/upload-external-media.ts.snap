// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`upload-external-media.ts: Uploads external media to media services 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "layout": "center",
        "width": null,
      },
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": "DUMMY-OBJECT-ID",
            "__displayType": null,
            "__fileMimeType": null,
            "__fileName": "example.png",
            "__fileSize": null,
            "collection": "MediaServicesSample",
            "height": 619,
            "id": "1234-5678-abcd-efgh",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": 1100,
          },
          "type": "media",
        },
      ],
      "type": "mediaSingle",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
